aiohttp/client.py

```diff
@@ -363,7 +378,10 @@ async def _request(
         proxy_headers = self._prepare_headers(proxy_headers)
 
         try:
-            url = URL(str_or_url)
+            if isinstance(str_or_url, URL):
+                url = str_or_url
+            else:
+                url = self._build_url(str_or_url)
```

Comment:
```suggestion
            url = self._build_url(URL(str_or_url))
```
URL.join() supports URL argument, this version can handle relative URLs (`URL('rel/path/to?a=1&b=2')`) as a bonus.

Reply:
Already done